### ISSUE TITLE
Flash does not work on IIGS ROM1 or ROM3 - always stuck at 4%

### ISSUE THREAD

**Issue by chuck424242** [freitz85/AppleIISd#4]
I have two AppleIISD cards and wanted to update to 1.2.2 (currently I am on 1.2.1)
I tried the flasher, put the jumper to R/W (2-3). It will start to flash, but always gets stuck at 4%

Tried it on a ROM1 and ROM3 IIGS, in 1MHz and 2.8MHz speeds. Same results on both machines.

**Comment by freitz85** [comment]
I have had the same issue on two boards but only on the IIe, the IIgs seemed more reliable.
You could try to erase the EEPROM first by specifying "0" as the slot in the flasher app.

**Comment by chuck424242** [comment]
I have tried it, but the erase gets stuck at 1%. I tried again with flashing and it is still the same, stuck at 4%. I guess in the meanwhile I need to unsolder the 28F64 and programm it external. :(

**Comment by chuck424242** [comment]
Found something interessting. It is wokring ONLY when a SD Card is inserted. If NO SD Card is in. Flashing will always fail.

**Comment by freitz85** [comment]
This is indeed interesting, as the SD card should have no effect at all. The only connection between the part that accesses the SD card and the EEPROM is the write enable bit set in the CPLD registers. Apart from that all address translation for SLOTROM and the $C800 space is completely separate.

**Comment by chuck424242** [comment]
Yes, this was surprising to me as well. I did have the SD Cards removed just to be sure there is no interference. I have test a couple of more times, and it is reproduceable, as soon as the SD card is in it always works. As soon as no SD card is in it fails at 1% for erase and 4% for flashing. I have just tested with the 1.2.2. not with 1.2.1 yet

**Comment by freitz85** [comment]
Thank you for verifying this behaviour.
I will try to test this and review the VHDL code if some strange interaction has crept in.

If you like to, you could test it with the previous version of the flasher application to see if the behaviour stays the same.
